370-auth-ui/oidc.md

```diff
@@ -0,0 +1,455 @@
+
+The values will be `["code", "none"]`.
+
+#### grant_types_supported
+
+The values will be `["authorization_code", "refresh_token"]`.
+
+#### subject_types_supported
+
+The values will be `["public"]`.
+
+#### id_token_signing_alg_values_supported
+
+`RS256` is supported.
+
+#### claims_supported
+
+The values will be `["sub", "iss", "aud", "exp", "iat"]`.
+
+### code_challenge_methods_supported
+
+The values will be `["plain", "S256"]`
+
+### revocation_endpoint
+
+No difference from the spec. See [Revocation Request](https://tools.ietf.org/html/rfc7009#section-2.1) for detail.
+
+## Signing ID token
+
+ID token signing key will be generated per app during app creation and store in App Config. For config compatibility, keys is an array, but Auth Gear only supports one key in this stage. Keys length will be checked by JSON schema.
+
+```yaml
+app_config:
+  oidc:
+    keys:
+    - kid: key_1
+      public_key: ""
+      private_key: ""
+```
+
+## Configurations
+
+### Sample configuration
+
+```yaml
+app_config:
+  clients:
+  - client_id: XXX
+    client_secret: XXX
+    refresh_token_lifetime: 86400
+    access_token_lifetime: 1800
+    session_idle_timeout_enabled: true
+    session_idle_timeout: 300
+    client_name: Mobile App
+    redirect_uris:
+    - "https://example.com"
+    logo_uri: "https://example.com/logo.png"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+### Parameters
+
+#### OIDC client Metadata
+
+Some parameters are defined in OIDC, see [ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata).
+
+They are
+
+- `client_name`
+- `redirect_uris`
+- `login_uri`
+- `grant_types`
+- `response_types`
+
+#### Skygear-specific
+
+- `client_id`: OIDC client id.
+- `client_secret`: OIDC client secret, used in RP integration.
+- `access_token_lifetime`: OIDC access token lifetime in seconds, default to 1800.
+- `refresh_token_lifetime`: OIDC refresh token lifetime in seconds, default to max(access_token_lifetime, 86400). Must greater than or equal to `access_token_lifetime`.
+- `session_idle_timeout_enabled`: Indicate whether OIDC session idle timeout is enabled, default to `false`.
+- `session_idle_timeout`: The OIDC session idle timeout in seconds, default to min(`access_token_lifetime`, 300). Must less than or equal to `access_token_lifetime`.
+
+### Configuration in different cases
+
+#### Generic OIDC RP
+
+```yaml
+app_config:
+  clients:
+  - redirect_uris:
+    - "https://app-backend-endpoint.com"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+- Should include app backend endpoint in `redirect_uris` which handle code to access token exchange
+- Generic OIDC RP run authentication code flow, and token endpoint will return refresh token. So `grant_types` should be [`authorization_code`, `refresh_token`] and `response_types` should be [`code`].
```

Comment:
The OIDC flow should be 'Authorization' Code flow, please use the correct name for the other cases as well.